### DISCUSSION TITLE
DeepAR very poor forecasts for weekly data

### DISCUSSION THREAD

**Discussion by mariosantosprivate** [awslabs/gluonts#1189] [Q&A]
Hi, I'm training a Deep AR model on a dataset with 1800 time series of weekly sales, using both stat_cat_feat (product, store, region etc) and dynamic_real_feats (promotions, out_of_stock). The original dataset consisted of daily sales and I aggregated to weekly to minimize the ammount of zeros.

Using the negative binomial distribution with the original data (daily sales) I got average results, MASE = 0.66, but after aggregating them the forecasts are completely off, and MASE shot up to 10.

[
![Untitled](https://user-images.githubusercontent.com/25696878/100777187-61791600-33fd-11eb-9627-0122b3fcf231.png)
](URL)

Unfortunately I cannot share the dataset, the following is the relevant code:

(freq is set to W-MON as the start date is a Monday and the data is weekly)
```python
train_ds = ListDataset([
    {
        FieldName.TARGET: target,
        FieldName.START: start,
        FieldName.FEAT_DYNAMIC_REAL: fdr,
        FieldName.FEAT_STATIC_CAT: fsc
    }
    for (target, start, fdr, fsc) in zip(train_target_values,
                                         dates,
                                         train_dynamic_features_list,
                                         stat_cat)
], freq="W-MON")

val_ds = ListDataset([
    {
        FieldName.TARGET: target,
        FieldName.START: start,
        FieldName.FEAT_DYNAMIC_REAL: fdr,
        FieldName.FEAT_STATIC_CAT: fsc
    }
    for (target, start, fdr, fsc) in zip(val_target_values,
                                         dates,
                                         val_dynamic_features_list,
                                         stat_cat)
], freq="W-MON")

test_ds = ListDataset([
    {
        FieldName.TARGET: target,
        FieldName.START: start,
        FieldName.FEAT_DYNAMIC_REAL: fdr,
        FieldName.FEAT_STATIC_CAT: fsc
    }
    for (target, start, fdr, fsc) in zip(test_target_values,
                                         dates,
                                         test_dynamic_features_list,
                                         stat_cat)
], freq="W-MON")

deepAR_estimator = DeepAREstimator(
    prediction_length=prediction_length,
    context_length=prediction_length,
    freq="W",
    num_layers=5,
    num_cells=40,
    distr_output = NegativeBinomialOutput(),
    use_feat_dynamic_real=True,
    use_feat_static_cat=True,
    cardinality=stat_cat_cardinalities,
    dropout_rate=0.1,
    
    trainer=Trainer(   
        learning_rate=1e-3,
        epochs=200,
        num_batches_per_epoch=100,
        batch_size=64
    )
)

deepAR_predictor = deepAR_estimator.train(training_data=train_ds, validation_data=val_ds)
```

What might be causing this?
And an unrelated question to which I could not find the answer in the documentation, what is the default loss function being used? 

Edit:
Training also outputs the following warning:
> weekofyear and week have been deprecated, please use DatetimeIndex.isocalendar().week instead, which returns a Series.  To exactly reproduce the behavior of week and weekofyear and return an Index, you may call pd.Int64Index(idx.isocalendar().week)

**Comment by lostella** [comment]
@mariosantosprivate as a test, I suggest trying out a different output distribution, for example leaving the default one (student-t). In other cases it was observed that negative binomial might be hard to fit well. Another option for integer data is to use a `PoissonOutput`.

The loss being optimized is negative log-likelihood.

**Comment by mariosantosprivate** [comment]
@lostella Thank you, changing the distribution to default (student T) fixed it, the forecasts are no longer ridiculous. They're still not very accurate but I believe the cause for that may lie in the data itself. Mainly nominal categorical features with very high cardinality which were simply label encoded since one-hot in these cases would explode the memory. 
Do you have any suggestions for such cases?
